src/account.ts

```diff
@@ -144,28 +112,55 @@ export class Account {
     protected async signAndSendTransaction(receiverId: string, actions: Action[]): Promise<FinalExecutionOutcome> {
         await this.ready;
 
-        // TODO: Find matching access key based on transaction
-        const accessKey = await this.findAccessKey();
-        if (!accessKey) {
-            throw new TypedError(`Can not sign transactions for account ${this.accountId}, no matching key pair found in Signer.`, 'KeyNotFound');
-        }
+        let txHash, signedTx;
+        // TODO: TX_NONCE (different constants for different uses of exponentialBackoff?)
+        const result = await exponentialBackoff(TX_STATUS_RETRY_WAIT, TX_NONCE_RETRY_NUMBER, TX_STATUS_RETRY_WAIT_BACKOFF, async () => {
+            const accessKeyInfo = await this.findAccessKey(receiverId, actions);
+            if (!accessKeyInfo) {
+                throw new TypedError(`Can not sign transactions for account ${this.accountId}, no matching key pair found in Signer.`, 'KeyNotFound');
+            }
+            const { publicKey, accessKey } = accessKeyInfo;
 
-        const status = await this.connection.provider.status();
+            const status = await this.connection.provider.status();
 
-        const [txHash, signedTx] = await signTransaction(
-            receiverId, ++accessKey.nonce, actions, base_decode(status.sync_info.latest_block_hash), this.connection.signer, this.accountId, this.connection.networkId
-        );
+            const nonce = ++accessKey.nonce;
+            [txHash, signedTx] = await signTransaction(
+                receiverId, nonce, actions, base_decode(status.sync_info.latest_block_hash), this.connection.signer, this.accountId, this.connection.networkId
+            );
+
+            try {
+                const result = await exponentialBackoff(TX_STATUS_RETRY_WAIT, TX_STATUS_RETRY_NUMBER, TX_STATUS_RETRY_WAIT_BACKOFF, async () => {
+                    try {
+                        return await this.connection.provider.sendTransaction(signedTx);
```

Comment:
If I understand this logic correctly, we should not retry that frequently, especially at the beginning of the backoff.

Reply:
@bowenwang1996 why? We are only doing new `broadcast_tx_commit` call after previous one returned timeout (my understanding is that it takes 10 seconds).

Reply:
ah I got confused.